internal/k8s/reconciler/gateway.go

```diff
@@ -330,15 +386,14 @@ func (g *K8sGateway) Status() gw.GatewayStatus {
 	}
 
 	ipType := gw.IPAddressType
-	addresses := []gw.GatewayAddress{}
+	addresses := make([]gw.GatewayAddress, 0, len(g.addresses))
 	for _, address := range g.addresses {
 		addresses = append(addresses, gw.GatewayAddress{
 			Type:  &ipType,
 			Value: address,
 		})
 	}
 
-	// TODO: set addresses based off of pod/service lookup
```

Comment:
I _believe_ this PR addresses this TODO, but lemme know if you don't think it does.

Reply:
Yep, I had actually forgotten I had thrown this comment in here back in October, so I guess I realized the shortcomings of the current approach back then but amidst additional implementations and prepping for releases totally forgot about this 😅